go/engine/list_tracking.go

```diff
@@ -59,149 +54,123 @@ func (e *ListTrackingEngine) RequiredUIs() []libkb.UIKind { return []libkb.UIKin
+	} else {
+		useServerLookup = true
+		m.Debug("got following list from server (len=%d, cachedOnly=%t); using it to filter sigchain list", len(ts.Result().Users), e.arg.CachedOnly)
+		for _, user := range ts.Result().Users {
+			serverLookup[user.Uid] = struct{}{}
+			fullNames[user.Uid] = user.FullName
 		}
+	}
 
-		var trackList TrackList
-		var err error
-		if idTable := user.IDTable(); idTable != nil {
-			trackList = idTable.GetTrackList()
-		}
+	// Load unstubbed so we get track links
+	larg := libkb.NewLoadUserArgWithMetaContext(m).
+		WithUID(uid).
+		WithStubMode(libkb.StubModeUnstubbed).
+		WithCachedOnly(e.arg.CachedOnly).
+		WithSelf(uid.Exists() && uid.Equal(m.G().GetMyUID()))
+	upak, _, err := m.G().GetUPAKLoader().LoadV2(larg)
+	if err != nil {
+		return err
+	}
+	if upak == nil {
+		return libkb.UserNotFoundError{}
+	}
+	unfilteredTracks := upak.Current.RemoteTracks
 
-		trackList, err = filterRxx(trackList, e.arg.Filter)
+	var rxx *regexp.Regexp
+	if e.arg.Filter != "" {
+		rxx, err = regexp.Compile(e.arg.Filter)
 		if err != nil {
 			return err
 		}
-		sort.Sort(trackList)
-
-		if e.arg.JSON {
-			return e.runJSON(m, trackList, e.arg.Verbose)
-		}
-		return e.runTable(m, trackList)
-	})
-
-	return err
-}
-
-func filterTracks(trackList TrackList, f func(libkb.TrackChainLink) bool) TrackList {
-	ret := TrackList{}
-	for _, link := range trackList {
-		if f(*link) {
-			ret = append(ret, link)
-		}
 	}
-	return ret
-}
 
-func filterRxx(trackList TrackList, filter string) (ret TrackList, err error) {
-	if len(filter) == 0 {
-		return trackList, nil
-	}
-	rxx, err := regexp.Compile(filter)
-	if err != nil {
-		return
-	}
-	return filterTracks(trackList, func(l libkb.TrackChainLink) bool {
-		if rxx.MatchString(l.ToDisplayString()) {
-			return true
-		}
-		for _, sb := range l.ToServiceBlocks() {
-			_, v := sb.ToKeyValuePair()
-			if rxx.MatchString(v) {
-				return true
+	// Filter out any marked reset by server, or due to Filter argument
+	var filteredTracks []keybase1.RemoteTrack
+	for _, track := range unfilteredTracks {
+		trackedUID := track.Uid
+		if useServerLookup {
+			if _, ok := serverLookup[trackedUID]; !ok {
+				m.Debug("filtering out uid %s in sigchain list but not provided by server", trackedUID)
+				continue
 			}
 		}
-		return false
-	}), nil
-}
-
-func (e *ListTrackingEngine) linkPGPKeys(m libkb.MetaContext, link *libkb.TrackChainLink) (res []keybase1.PublicKey) {
-	trackedKeys, err := link.GetTrackedKeys()
-	if err != nil {
-		m.Warning("Bad track of %s: %s", link.ToDisplayString(), err)
-		return res
+		if rxx != nil && !rxx.MatchString(track.Username) {
+			continue
+		}
+		filteredTracks = append(filteredTracks, track)
 	}
 
-	for _, trackedKey := range trackedKeys {
-		res = append(res, keybase1.PublicKey{PGPFingerprint: trackedKey.Fingerprint.String()})
+	sort.Slice(filteredTracks, func(i, j int) bool {
+		return strings.Compare(filteredTracks[i].Username, filteredTracks[j].Username) < 0
```

Comment:
You probably don't want `strings.Compare` https://stackoverflow.com/questions/52830314/go-compare-strings/52830378
Simple `<` is better

Reply:
TIL, thanks. That's odd.